### PR TITLE
List only a single key for S3 healthcheck

### PR DESCRIPTION
**Pull request by bdurrani**
**What this PR does / why we need it**: See issues #1705, which details why this PR is a good idea.

**Which issue(s) this PR fixes**: #1705 

Please reference the issue this PR will close: #1705 

**Special notes for your reviewer**: Unsure how to unit test this change

**Does this PR introduce a user-facing change?**: No

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

**Comment by sungam3r** [comment]
I see a lot of whitespace changes. Please revert all of them. Maybe you need to configure your git client - `autocrlf` option or something like that. After that I'm fine to merge.

File: src/HealthChecks.Aws.S3/S3HealthCheck.cs [hunk 1]
```diff
@@ -14,39 +15,40 @@ public S3HealthCheck(S3BucketOptions bucketOptions)
...
-
-                if (_bucketOptions.CustomResponseCheck != null)
-                {
-                    return _bucketOptions.CustomResponseCheck.Invoke(response)
-                        ? HealthCheckResult.Healthy()
+                : new AmazonS3Client(_bucketOptions.S3Config);
+
+            using (client)
+            {
+                var listRequest = new ListObjectsRequest { BucketName = _bucketOptions.BucketName, MaxKeys = 1 };
```
Comment (sungam3r) [inline src/HealthChecks.Aws.S3/S3HealthCheck.cs:45]: Let's move that setting into `S3BucketOptions ` with default value 1.

**Comment by sungam3r** [comment]
#1718

**Comment by bdurrani** [comment]
@sungam3r PR is ready for your review

**Review by sungam3r — APPROVED** [review]

**Pull request merged**